scripts/prepdocs.py

```diff
@@ -74,18 +96,74 @@ def remove_blobs(filename):
             if args.verbose: print(f"\tRemoving blob {b}")
             blob_container.delete_blob(b)
 
-def split_text(pages):
+def table_to_html(table):
+    html = "<table>"
+    rows = [sorted([cell for cell in table.cells if cell.row_index == i], key=lambda cell: cell.column_index) for i in range(table.row_count)]
+    for row_cells in rows:
+        html += "<tr>"
+        for cell in row_cells:
+            tag = "th" if (cell.kind == "columnHeader" or cell.kind == "rowHeader") else "td"
+            cell_spans = ""
+            if cell.column_span > 1: cell_spans += f" colSpan={cell.column_span}"
+            if cell.row_span > 1: cell_spans += f" rowSpan={cell.row_span}"
+            html += f"<{tag}{cell_spans}>{cell.content}</{tag}>"
+        html +="</tr>"
+    html += "</table>"
+    return html
+
+def get_document_text(filename):
+    offset = 0
+    page_map = []
+    if args.localpdfparser:
+        reader = PdfReader(filename)
+        pages = reader.pages
```

Comment:
A bit unfortunate that we're opening/parsing each PDF twice (here and in upload_blobs). No the end of the world if it'll keep the code simple since this is an example.

Reply:
Yes, I've seen this issue but I've left it as is for now to make code simpler.